engine/src/graphics/device.cpp

```diff
@@ -59,13 +69,103 @@ namespace gen
+        for (int i = 0; i < availablePhysicalDevices.size(); i++)
+        {
+            aDev << "\tDevice [" << i << "] : " << "\n"
+                 << "\t\tName: " << availablePhysicalDevices[i].getProperties().deviceName << "\n"
+                 << "\t\tType: " << to_string(availablePhysicalDevices[i].getProperties().deviceType) << "\n"
+                 << "\t\tAPI Version: " << availablePhysicalDevices[i].getProperties().apiVersion << "\n"
+                 << "\t\tDriver Version: " << availablePhysicalDevices[i].getProperties().driverVersion << "\n"
+                 << "\t\tVendor ID: " << availablePhysicalDevices[i].getProperties().vendorID << "\n"
+                 << "\t\tDevice ID: " << availablePhysicalDevices[i].getProperties().deviceID
+                 << "\n";
+        }
+        gen::logger::debug("vulkan", std::format("Found Physical Devices: \n{}\n", aDev.str()));
+	}
+
+	void GraphicsDevice::createLogicalDevice()
+    {
+        QueueFamilyIndices indices = findQueueFamilies(m_physicalDevice);
+
+        std::vector<vk::DeviceQueueCreateInfo> queueCreateInfos;
+        std::set<uint32_t> const uniqueQueueFamilies = {indices.graphicsFamily.value()};
+
+        float const queuePriority = 1.0F;
+        for (u32 const queueFamily : uniqueQueueFamilies)
+        {
+            vk::DeviceQueueCreateInfo const queueCreateInfo({}, queueFamily, 1, &queuePriority);
+            queueCreateInfos.push_back(queueCreateInfo);
+        }
+
+        vk::PhysicalDeviceFeatures const deviceFeatures{}; // This is a placeholder for later.
 
+        vk::DeviceCreateInfo const createInfo({}, queueCreateInfos.size(), queueCreateInfos.data());
 
+        m_device = m_physicalDevice.createDeviceUnique(createInfo);
+
+        m_graphicsQueue = m_device->getQueue(indices.graphicsFamily.value(), 0);
+    }
+
+	QueueFamilyIndices GraphicsDevice::findQueueFamilies(vk::PhysicalDevice device)
+	{
+		QueueFamilyIndices indices;
+
+		auto queueFamilies = device.getQueueFamilyProperties();
+
+		auto qProp = std::stringstream{"vulkan"};
+		for (int i = 0; i < queueFamilies.size(); i++)
+		{
+			qProp << "\t" << "Queue Family [" << i << "] :\n"
+				  << "\t\t\tQueue Flags: " << to_string(queueFamilies[i].queueFlags) << "\n"
+				  << "\t\t\tQueue Count: " << queueFamilies[i].queueCount << "\n"
+				  << "\t\t\tTimestamp Valid Bits: " << queueFamilies[i].timestampValidBits << "\n"
+				  << "\t\t\tMin Image Transfer Granularity: width " << queueFamilies[i].minImageTransferGranularity.width << ", height "
+				  << queueFamilies[i].minImageTransferGranularity.height << " , depth "
+				  << queueFamilies[i].minImageTransferGranularity.depth
+				  << "\n";
+		}
+		gen::logger::debug("vulkan", std::format("Found Queue Family Properties: \n{}\n", qProp.str()));
+
+		int index = 0;
+		for (const auto & queueFamily : queueFamilies)
+		{
+			if (queueFamily.queueFlags & vk::QueueFlagBits::eGraphics)
+			{
+				indices.graphicsFamily = index;
+			}
```

Comment:
This is not enough, we need graphics, transfer, and present capabilities. Implementations tend to return the queue family that supports everything as the first one (which is why "it works"), but that's not guaranteed by the spec.